### DISCUSSION TITLE
Documentation for MAGICK* routines?

### DISCUSSION THREAD

**Discussion by jtappin** [gnudatalanguage/gdl#1125] [Q&A] [answered]
Since it is clear that `write_tiff` is, to put it relatively politely, a train wreck, I've started to have a hack at it. 
Is there a list of available MAGICK* routines and their arguments & keywords anywhere?

**Comment by GillesDuvert** [comment, marked as answer]
Not really, but the functions and procedures are called according to what is in libinit_cl.cpp
The code is of course in magick_cl.cpp
one can have a summary like 
```
GDL> #magick_read
Internal FUNCTION : res=MAGICK_READ([1 Arg],RGB,SUB_RECT,MAP)
```
This is very rudimentary indeed

**Reply by jtappin** [reply]
I'll have a look. Anyhow, I think I've got a new write_tiff that does what the old one did but with two important improvements:
1. It actually honours colour tables for pseudo-colour images
2. It works with 3-D images provided one dimension is of length 3.

I'll make an issue and attach a copy.
